server/src/test/java/io/druid/segment/realtime/appenderator/FiniteAppenderatorDriverTest.java

```diff
@@ -155,21 +162,52 @@ public void testMaxRowsPerSegment() throws Exception
       InputRow row = new MapBasedInputRow(
           new DateTime("2000T01"),
           ImmutableList.of("dim2"),
-          ImmutableMap.<String, Object>of(
+          ImmutableMap.of(
               "dim2",
               String.format("bar-%d", i),
               "met1",
               2.0
           )
       );
-      Assert.assertNotNull(driver.add(row, "dummy", committerSupplier));
+      final AppenderatorDriverAddResult addResult = driver.add(row, "dummy", committerSupplier);
+      Assert.assertTrue(addResult.isOk());
+      if (addResult.getNumRowsInSegment() > MAX_ROWS_PER_SEGMENT) {
+        driver.moveSegmentOut("dummy", ImmutableList.of(addResult.getSegmentIdentifier()));
+      }
     }
 
-    final SegmentsAndMetadata segmentsAndMetadata = driver.finish(makeOkPublisher(), committerSupplier.get());
+    final SegmentsAndMetadata published = driver.publish(
+        makeOkPublisher(),
+        committerSupplier.get(),
+        ImmutableList.of("dummy")
+    ).get(PUBLISH_TIMEOUT, TimeUnit.MILLISECONDS);
+    final SegmentsAndMetadata segmentsAndMetadata = driver.registerHandoff(published)
+                                                          .get(HANDOFF_CONDITION_TIMEOUT, TimeUnit.MILLISECONDS);
     Assert.assertEquals(numSegments, segmentsAndMetadata.getSegments().size());
     Assert.assertEquals(numSegments * MAX_ROWS_PER_SEGMENT, segmentsAndMetadata.getCommitMetadata());
   }
 
+  @Test(timeout = 5000L, expected = TimeoutException.class)
```

Comment:
In the case of no message, I think there's no reason to check the error message.